plugins/modules/docker_image.py

```diff
@@ -549,9 +550,26 @@ def archive_image(self, name, tag):
             self.log("archive image: image %s not found" % image_name)
             return
 
+        # Will have a 'sha256:' prefix
+        new_image_id = image['Id']
+
+        try:
+            archived_id = archived_image_id(self.archive_path)
+
+            if archived_id:
+                self.results['actions'].append('Previous image ID was %s' % archived_id)
```

Comment:
OK.

What are the differences between `self.log` and `module.debug` ?

I have no opinion either way, but want to understand them better.

Reply:
`self.log()` is the preferred way for the Docker modules, since all other Docker modules also use this. It does nothing, but can easily be enable to print something when debugging a module in the collection.

Reply:
[This line](https://github.com/ansible-collections/community.docker/pull/500/files#diff-6178e5dc28781ab3227b344e50384dafb10d5df4aa0b54dc6fa3585c5c259d14R549) is the only thing logged now:

It logs an exception, if any, from reading or parsing the tar file:

```
'Unable to extract manifest summary from archive: %s' % to_native(exc)
```

How do I call `module.debug` ?

That doesn't resolve in my IDE.

Reply:
Did you look in https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1305 ?